src/promptflow-azure/promptflow/azure/operations/_arm_connection_operations.py

```diff
@@ -43,8 +45,29 @@ def __init__(
             self._credential,
         )
 
+    @classmethod
+    def _convert_core_connection_to_sdk_connection(cls, core_conn):
+        # TODO: Refine this and connection operation ones to (devkit) _Connection._from_core_object
+        sdk_conn_mapping = _Connection.SUPPORTED_TYPES
+        sdk_conn_cls = sdk_conn_mapping.get(core_conn.type)
+        if sdk_conn_cls is None:
+            raise ConnectionClassNotFoundError(
+                f"Correspond sdk connection type not found for core connection type: {core_conn.type!r}, "
+                f"please re-install the 'promptflow' package."
+            )
+        common_args = {
+            "name": core_conn.name,
+            "module": core_conn.module,
+            "expiry_time": core_conn.expiry_time,
+            "created_date": core_conn.created_date,
+            "last_modified_date": core_conn.last_modified_date,
+        }
+        if sdk_conn_cls is CustomConnection:
+            return sdk_conn_cls(configs=core_conn.configs, secrets=core_conn.secrets, **common_args)
+        return sdk_conn_cls(**dict(core_conn), **common_args)
```

Comment:
This may cause errors for multiple keys?
![image](https://github.com/microsoft/promptflow/assets/23182548/2372af9a-e6c2-428f-b28a-15b9026cb860)

Reply:
dict(core_conn) will not return name.
![image](https://github.com/microsoft/promptflow/assets/24237253/b2235b76-a2e4-41d3-b81f-ebf320bd91fd)